service/framework.go

```diff
@@ -0,0 +1,419 @@
+	{
+		c := serviceresource.DefaultConfig()
+
+		c.K8sClient = k8sClient
+		c.Logger = config.Logger
+
+		serviceResource, err = serviceresource.New(c)
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+	}
+
+	var resources []framework.Resource
+	{
+		resources = []framework.Resource{
+			namespaceResource,
+
+			configMapResource,
+			deploymentResource,
+			ingressResource,
+			pvcResource,
+			serviceResource,
+		}
+
+		logWrapConfig := logresource.DefaultWrapConfig()
+
+		logWrapConfig.Logger = config.Logger
+
+		resources, err = logresource.Wrap(resources, logWrapConfig)
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+
+		retryWrapConfig := retryresource.DefaultWrapConfig()
+
+		retryWrapConfig.BackOffFactory = func() backoff.BackOff { return backoff.WithMaxTries(backoff.NewExponentialBackOff(), ResourceRetries) }
+		retryWrapConfig.Logger = config.Logger
+
+		resources, err = retryresource.Wrap(resources, retryWrapConfig)
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+
+		metricsWrapConfig := metricsresource.DefaultWrapConfig()
+
+		metricsWrapConfig.Name = config.Name
+
+		resources, err = metricsresource.Wrap(resources, metricsWrapConfig)
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+	}
+
+	var newTPR *tpr.TPR
+	{
+		c := tpr.DefaultConfig()
+
+		c.K8sClient = k8sClient
+		c.Logger = config.Logger
+
+		c.Description = kvmtpr.Description
+		c.Name = kvmtpr.Name
+		c.Version = kvmtpr.VersionV1
+
+		newTPR, err = tpr.New(c)
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+	}
+
+	var newWatcherFactory informer.WatcherFactory
+	{
+		zeroObjectFactory := &informer.ZeroObjectFactoryFuncs{
+			NewObjectFunc:     func() runtime.Object { return &kvmtpr.CustomObject{} },
+			NewObjectListFunc: func() runtime.Object { return &kvmtpr.List{} },
+		}
+		newWatcherFactory = informer.NewWatcherFactory(k8sClient.Discovery().RESTClient(), newTPR.WatchEndpoint(""), zeroObjectFactory)
+	}
+
+	var newInformer *informer.Informer
+	{
+		c := informer.DefaultConfig()
+
+		c.WatcherFactory = newWatcherFactory
+
+		c.RateWait = 10 * time.Second
+		c.ResyncPeriod = 5 * time.Minute
+
+		newInformer, err = informer.New(c)
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+	}
+
+	initCtxFunc := func(ctx context.Context, obj interface{}) (context.Context, error) {
+		ctx = messagecontext.NewContext(ctx, messagecontext.NewMessage())
+
+		return ctx, nil
+	}
+
+	var customObjectFramework *framework.Framework
+	{
+		c := framework.DefaultConfig()
+
+		c.BackOffFactory = framework.DefaultBackOffFactory()
+		c.Informer = newInformer
+		c.InitCtxFunc = initCtxFunc
+		c.Logger = config.Logger
+		c.ResourceRouter = framework.DefaultResourceRouter(resources)
+		c.TPR = newTPR
+
+		customObjectFramework, err = framework.New(c)
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+	}
+
+	return customObjectFramework, nil
+}
+
+func newPodFramework(config Config) (*framework.Framework, error) {
```

Comment:
This is the new additional pod framework. We make the kvm-operator able to watch multiple different resources. We want to make the operator also watching pods it creates itself. That way we can manage the deletion of the guest cluster pods within separated resource implementations and do the draining magic.